spec/views/posts/posts_index_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature, js: false do
+  before :each do
+    @user = User.create(name: 'Greg', photo: 'https://randomuser.me/api/portraits/men/2.jpg',
+                        bio: 'Teacher in Zimbabwe')
+    @first_post = Post.create(author: @user, title: 'Hello', text: 'This is my first post')
+    @comment = Comment.create(post: @first_post, author: @user, text: 'Hi Tom!')
+
+    visit "/users/#{@user.id}/posts"
+  end
+
+  describe 'Post page' do
+    it "shows the user's profile" do
+      expect(page).to have_css("img[src*='https://randomuser.me/api/portraits/men/2.jpg']")
+    end
+
+    it "shows the user's username" do
+      Post.find_by(author: @user.name)
+      expect(page).to have_content(@user.name)
+    end
+
+    it 'shows the number of post' do
+      expect(page).to have_content(@user.post_counter)
+    end
+
```

Comment:
- [ ] As mentioned by the previous reviewer, You missed the below tests for the post index page from the requirements. Please make sure to add them and they should pass without any errors.
  - `I can see a section for pagination if there are more posts than fit on the view.`